src/models/model.py

```diff
@@ -16,6 +16,32 @@
 
 from pytorch_lightning.metrics.functional import ssim, psnr
 from pytorch_lightning.callbacks.early_stopping import EarlyStopping
+from pytorch_lightning.callbacks import Callback
+
+
+class SavePredImgCallback(Callback):
+    def __init__(self, dl, epoch_logging_freq: int = 50):
+        # save every 50 epochs
```

Comment:
not sure if this is too frequent? basically i want to ability to manually monitor and shut down any jobs but still have the predictions on those same val predictions and if i do every 100 epochs the last prediction might be too far back?

Reply:
thoughts on this?

Reply:
Hmm, might be best to save the best model weights and load the model for generating validation images (I know this is a different paradigm that what we have been using).

Reply:
i did start with this but its also quite slow/frustrating to do this for 10 models so kind of wanted to generate automatically - I'll think about this again now